### PR TITLE
Swc-1954: Expose way to change project/folder storage location

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/controller/StorageLocationWidget.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+	@Override
+	public void onSave() {
+		synAlert.clear();
+		StorageLocationSetting setting = getStorageLocationSettingFromView();
+		String error = validate(setting);
+		if (error != null) {
+			synAlert.showError(error);
+		} else {
+			//look for duplicate storage location in existing settings
```
Comment (marcel-blonk) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/controller/StorageLocationWidget.java:121]: This comment and the code don't seem to agree?

File: src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java [hunk 2]
```diff
@@ -3095,4 +3063,89 @@ public EntityBundlePlus getEntityInfo(String entityId) throws RestServiceExcepti
 		}
 	}
 	
+	@Override
+	public List<String> getMyLocationSettingBanners() throws RestServiceException{
+		try {
+			Set<String> banners = new HashSet<String>();
```
Comment (marcel-blonk) [inline src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java]: Use treeset and get sorting automatically? (have to add case insensitve comparator)

File: src/main/java/org/sagebionetworks/web/client/widget/entity/controller/StorageLocationWidget.java (hunk 1 above)
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/controller/StorageLocationWidget.java:121]: Good catch. I ended up doing this servlet side instead, see SynapseClientImpl.createStorageLocationSetting()

**Comment by jay-hodgson** [comment]
blessed by Marcel

**Pull request merged**